### PR TITLE
Improve init error handling

### PR DESCRIPTION
**Pull request by dfabulich**
This adds a new `init()` function to greenworks, which throws an exception with a clear error message when initialization fails (by applying standard heuristics).

This required exposing `SteamAPI_IsSteamRunning()` which is super useful; it can return `true` even if your `steam_appid.txt` is missing.

### CODE REVIEW

**Comment by patrickklug** [comment]
I'm a bit confused by this. It seems to mostly throw errors about missing `steam_appid.txt` but this file is not required, nor recommended for actual deployment.

**Comment by dfabulich** [comment]
The algorithm here is a series of checks:

1) If `initAPI` succeeds, return without throwing
2) If Steam isn't running, throw an exception saying so.
3) Read `steam_appid.txt`; throw if it's missing.
4) Verify that `steam_appid.txt` contains a simple numeric string; throw if it's invalid.
5) Throw a `shrug` exception if it's still failing.

I claim that this will do the right thing even in actual deployment, because we'll never get past step 2 in production.

https://partner.steamgames.com/documentation/getting_started

> A return of `false` indicates one of three conditions:
> - The Steam client isn't running. A running Steam client is required to provide implementations of the various Steamworks interfaces.
> - The Steam client couldn't determine the AppID of game. Make sure you have Steam_appid.txt in your game directory.
> - Your application is not running under the same user context as the Steam client, including admin privileges.
> 
> Note: When you launch from Steam, Steam will automatically know the AppID of your game. When developing, you need to hint this to Steam with a text file. Create the text file `steam_appid.txt` containing only the AppID.

As I read it (and as I understand it), `SteamAPI_Init()` should "never" fail in actual deployment, but it could happen if Steam launched the game but then died before `initAPI` was called. (In that case, the algorithm in this PR would throw at step 2.)

It should really _never_ be possible for `SteamAPI_Init()` to fail in actual deployment if `SteamAPI_IsSteamRunning()` returns true. `IsSteamRunning()` establishes a "steam pipe" connection to Steam itself, so if `IsSteamRunning` passes, then there's no way left for `Init` to fail in actual deployment.

Therefore, if `SteamAPI_IsSteamRunning() && !SteamAPI_Init()`, then we _must_ be running in development mode, and therefore there must be something wrong with `steam_appid.txt`.

(It is probably possible for `SteamAPI_IsSteamRunning()` to return `false` in actual deployment even if Steam _is_ running, e.g. if Steam is running as admin but the app isn't, but that will just make the algorithm halt on step 2; it won't complain about missing `steam_appid.txt` in that case.)

**Comment by patrickklug** [comment]
Thanks for the clarification!

**Comment by hokein** [comment]
Awesome! Looks good to me. Could you help to resolve the conflicts in the PR? If you don't have time, I can do it myself :-).

**Comment by dfabulich** [comment]
Rebased, conflicts resolved.

File: greenworks.js [hunk 1]
```diff
@@ -96,6 +96,22 @@ greenworks.Utils.move = function(source_dir, target_dir, success_callback,
...
+  var appId;
+  try {
+    appId = fs.readFileSync('steam_appid.txt', 'utf8');
+  } catch (e) {
+    throw new Error("Steam initialization failed. Steam is running, but steam_appid.txt is missing. Expected to find it in: " + require('path').resolve('steam_appid.txt'));
+  }
+  if (!/^\d+ *\r?\n?$/.test(appId)) {
+    throw new Error("Steam initialization failed. steam_appid.txt appears to be invalid; it should contain a numeric ID: " + appId);
+  }
+  var shrug = '\xAF\\_(\u30C4)_/\xAF';
```
Comment (hokein) [inline greenworks.js]: Could you explain the magic value of  `shrug`? I'm confessed at the first glance. Would be better to add a comment.

Comment (dfabulich) [inline greenworks.js]: The shrug is JavaScript-escaped string "¯\\\_(ツ)_/¯" and is meant to be a playful way of saying "we have no idea what could be wrong at this point."

The exception message looks like this on the console:
> Steam initialization failed, but Steam is running, and steam_appid.txt is present and valid. ¯\\\_(ツ)_/¯ Maybe that's not really YOUR app ID? 123456

(I think JS supports utf8 out of the box, so I could literally have written `var shrug = "¯\_(ツ)_/¯"` but I always prefer to write my code in ASCII wherever possible.)

I could certainly remove it if you'd prefer, but I like it.

Comment (hokein) [inline greenworks.js]: Aha, I see. Thanks for the explanation. It looks fine, but I'd not suggest to add the emoji in the error message.

**Comment by dfabulich** [comment]
Force pushed; rewrote the commit to use `:shrug:`

**Comment by hokein** [comment]
Looks good now, thanks!

**Pull request merged**